src/crud/document.py

```
@@ -670,13 +738,27 @@ async def create_documents(
             if doc.embedding:
                 docs_with_embeddings.append((new_doc, doc.embedding))
 
+        except SQLAlchemyError:
+            # The session/transaction is dead; continuing the loop would only
+            # cascade PendingRollbackErrors and lose the whole batch silently.
+            await db.rollback()
+            raise
         except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)